index.js

```diff
@@ -23,6 +91,102 @@ const scrollHandler = (event) => {
   }
 };
 
+const closePopUpHandler = (event) => {
+  event.preventDefault();
+  overly.classList.add('hidden-g');
+};
+
+const renderPopUp = (e) => {
+  e.preventDefault();
+  popups.innerHTML = '';
+  overly.classList.remove('hidden-g');
+  const id = e.target.dataset.show;
+  const [templateMakeup] = projectData.filter((item) => item.id === id);
+  const markUp = ` 
+  <div class="flex-row">
+  <h2>${templateMakeup.title}</h2>
+  <svg
+    xmlns="http://www.w3.org/2000/svg"
+    fill="none"
+    viewBox="0 0 24 24"
+    stroke-width="1.5"
+    height="32"
+    width="32"
+    stroke="#67798E"
+    class="w-6 h-6 crossIcon"
+  >
+    <path
+      stroke-linecap="round"
+      stroke-linejoin="round"
+      d="M6 18L18 6M6 6l12 12"
+    />
+  </svg>
+</div>
+<ul class="skills-lan flex-row">
+  <li><a class="tag" href="#">${templateMakeup.techStack.tech1}</a></li>
+  <li><a class="tag" href="#">${templateMakeup.techStack.tech2}</a></li>
+  <li><a class="tag" href="#">${templateMakeup.techStack.tech3}</a></li>
+</ul>
+<div class="popupContainer">
+  <div class="bg-img">
+    <img
+      class="projectsnap"
+      src=${templateMakeup.img}
+      alt="project screen shoot"
+    />
+  </div>
+  <div class="popupText">
+    <p>
+    ${templateMakeup.description}
+    </p>
+    <ul class="popup-btn flex-row">
+      <li class="flex-row popup-cont">
+        <a href="#">See live</a>
+        <img
+          class="popup-icon"
+          src="./style/imgs/Icon.png"
+          alt="live-share-icon"
+        />
+      </li>
+      <li class="flex-row popup-cont">
+        <a href="#">See source</a>
+        <img
+          class="popup-icon"
+          src="./style/imgs/Group.png"
+          alt="source code to github"
+        />
+      </li>
+    </ul>
+  </div>
+</div>
+  `;
+  popups.insertAdjacentHTML('afterbegin', markUp);
+  closePopUp = document.querySelector('.crossIcon');
+  closePopUp.addEventListener('click', closePopUpHandler);
+};
+
+const reset = () => {
+  userName.value = '';
+  email.value = '';
+  textBox.textContent = '';
+};
+
+const formSubmitHandler = (e) => {
+  e.preventDefault();
+  if (/^[a-z0-9._-]+@[a-z0-9.-]+\.[a-z]{2,4}$/g.test(email.value)) {
+    massageEmail.textContent = '';
+    form.submit();
+    messageValid.textContent = success;
+    reset();
+  } else {
+    massageEmail.textContent = invalidEmail;
+  }
+};
+
+form.addEventListener('submit', formSubmitHandler);
```

Comment:
Great work by trying to implement this functionality , but as requested in this project requirements the form should not be submitted if the regex failed to test but in your case it gets submitted in both with lowercase and upercase caracters , also I don't see the error or success messages while they should appear near the submit button you can set a timeOut before submitting the form to let the user see it